diff --git a/shard.yml b/shard.yml
index 5382bfa..228bf45 100644
--- a/shard.yml
+++ b/shard.yml
@@ -15,24 +15,27 @@ crystal: ~> 1.0
 dependencies:
   # annotation:
   #   github: grottopress/annotation.cr
-  #   version: ~> 0.4.0
+  #   version: ~> 0.5.0
   lucky:
     github: luckyframework/lucky
     version: ~> 0.27.0
   carbon:
     github: luckyframework/carbon
-    version: ~> 0.1.4
+    version: ~> 0.2.0
+  carbon_sendgrid_adapter:
+    github: luckyframework/carbon_sendgrid_adapter
+    version: ~> 0.1.0
   envy:
     github: GrottoPress/envy
     version: ~> 0.3.0
   shield:
     github: GrottoPress/shield
-    version: ~> 0.9.0
+    version: ~> 0.10.0
 
 development_dependencies:
   # ameba:
   #   github: veelenga/ameba
-  #   version: ~> 0.13.0
+  #   version: ~> 0.14.0
   lucky_flow:
     github: luckyframework/lucky_flow
     version: ~> 0.7.3
diff --git a/src/actions/api/bearer_logins/index.cr b/src/actions/api/bearer_logins/index.cr
index 1beb009..4374853 100644
--- a/src/actions/api/bearer_logins/index.cr
+++ b/src/actions/api/bearer_logins/index.cr
@@ -7,9 +7,7 @@ class Api::BearerLogins::Index < ApiAction
     json({
       status: "success",
       data: {
-        bearer_logins: BearerLoginSerializer.for_collection(
-          active_bearer_logins
-        )
+        bearer_logins: BearerLoginSerializer.for_collection(bearer_logins)
       },
       pages: {
         current: page,
@@ -17,8 +15,4 @@ class Api::BearerLogins::Index < ApiAction
       }
     })
   end
-
-  private def active_bearer_logins
-    bearer_logins.select &.active?
-  end
 end
diff --git a/src/actions/api/logins/index.cr b/src/actions/api/logins/index.cr
index 8cfe1ae..447041c 100644
--- a/src/actions/api/logins/index.cr
+++ b/src/actions/api/logins/index.cr
@@ -6,15 +6,11 @@ class Api::Logins::Index < ApiAction
   get "/logins" do
     json({
       status: "success",
-      data: {logins: LoginSerializer.for_collection(active_logins)},
+      data: {logins: LoginSerializer.for_collection(logins)},
       pages: {
         current: page,
         total: pages.total
       }
     })
   end
-
-  private def active_logins
-    logins.select &.active?
-  end
 end
diff --git a/src/actions/bearer_logins/index.cr b/src/actions/bearer_logins/index.cr
index 9fd55b1..bfb6eb4 100644
--- a/src/actions/bearer_logins/index.cr
+++ b/src/actions/bearer_logins/index.cr
@@ -4,10 +4,6 @@ class BearerLogins::Index < BrowserAction
   param page : Int32 = 1
 
   get "/bearer-logins" do
-    html IndexPage, bearer_logins: active_bearer_logins, pages: pages
-  end
-
-  private def active_bearer_logins
-    bearer_logins.select &.active?
+    html IndexPage, bearer_logins: bearer_logins, pages: pages
   end
 end
diff --git a/src/actions/logins/index.cr b/src/actions/logins/index.cr
index 116bfcc..07f9c8b 100644
--- a/src/actions/logins/index.cr
+++ b/src/actions/logins/index.cr
@@ -4,10 +4,6 @@ class Logins::Index < BrowserAction
   param page : Int32 = 1
 
   get "/logins" do
-    html IndexPage, logins: active_logins, pages: pages
-  end
-
-  private def active_logins
-    logins.select &.active?
+    html IndexPage, logins: logins, pages: pages
   end
 end
diff --git a/src/app.cr b/src/app.cr
index ffee3d7..7e2209c 100644
--- a/src/app.cr
+++ b/src/app.cr
@@ -8,6 +8,7 @@
 require "envy"
 require "lucky"
 require "carbon"
+require "carbon_sendgrid_adapter"
 # require "annotation"
 require "shield"
 
