diff --git a/.editorconfig b/.editorconfig
index 7eef262..6d64754 100644
--- a/.editorconfig
+++ b/.editorconfig
@@ -10,6 +10,3 @@ trim_trailing_whitespace = true
 
 [*.{diff,md}]
 trim_trailing_whitespace = false
-
-[*.md]
-indent_size = 4
diff --git a/shard.yml b/shard.yml
index 77448a9..b9d65f5 100644
--- a/shard.yml
+++ b/shard.yml
@@ -33,7 +33,7 @@ dependencies:
     version: ~> 0.2.1
   shield:
     github: GrottoPress/shield
-    version: ~> 0.15.0
+    version: ~> 0.16.0
   unslash:
     github: GrottoPress/unslash
     version: ~> 0.1.0
diff --git a/src/actions/api/bearer_logins/show.cr b/src/actions/api/bearer_logins/show.cr
new file mode 100644
index 0000000..589becd
--- /dev/null
+++ b/src/actions/api/bearer_logins/show.cr
@@ -0,0 +1,7 @@
+# class Api::BearerLogins::Show < ApiAction
+#   include Shield::Api::BearerLogins::Show
+
+#   get "/bearer-logins/:bearer_login_id" do
+#     json BearerLoginSerializer.new(bearer_login: bearer_login)
+#   end
+# end
diff --git a/src/actions/api/bearer_logins/verify.cr b/src/actions/api/bearer_logins/verify.cr
new file mode 100644
index 0000000..7714418
--- /dev/null
+++ b/src/actions/api/bearer_logins/verify.cr
@@ -0,0 +1,7 @@
+class Api::BearerLogins::Verify < ApiAction
+  include Shield::Api::BearerLogins::Verify
+
+  post "/bearer-logins/verify" do
+    run_operation
+  end
+end
diff --git a/src/actions/api/email_confirmations/show.cr b/src/actions/api/email_confirmations/show.cr
new file mode 100644
index 0000000..e26d5a6
--- /dev/null
+++ b/src/actions/api/email_confirmations/show.cr
@@ -0,0 +1,7 @@
+# class Api::EmailConfirmations::Show < ApiAction
+#   include Shield::Api::EmailConfirmations::Show
+
+#   get "/email-confirmations/:email_confirmation_id" do
+#     json EmailConfirmationSerializer.new(email_confirmation: email_confirmation)
+#   end
+# end
diff --git a/src/actions/api/logins/show.cr b/src/actions/api/logins/show.cr
new file mode 100644
index 0000000..0cdc789
--- /dev/null
+++ b/src/actions/api/logins/show.cr
@@ -0,0 +1,7 @@
+# class Api::Logins::Show < ApiAction
+#   include Shield::Api::Logins::Show
+
+#   get "/logins/:login_id" do
+#     json LoginSerializer.new(login: login)
+#   end
+# end
diff --git a/src/actions/api/password_resets/show.cr b/src/actions/api/password_resets/show.cr
new file mode 100644
index 0000000..f91c3ca
--- /dev/null
+++ b/src/actions/api/password_resets/show.cr
@@ -0,0 +1,7 @@
+# class Api::PasswordResets::Show < ApiAction
+#   include Shield::Api::PasswordResets::Show
+
+#   get "/password-resets/:password_reset_id" do
+#     json PasswordResetSerializer.new(password_reset: password_reset)
+#   end
+# end
diff --git a/src/actions/bearer_logins/show.cr b/src/actions/bearer_logins/show.cr
index cff88c4..e694bbf 100644
--- a/src/actions/bearer_logins/show.cr
+++ b/src/actions/bearer_logins/show.cr
@@ -1,9 +1,9 @@
-class BearerLogins::Show < BrowserAction
-  include Shield::BearerLogins::Show
+# class BearerLogins::Show < BrowserAction
+#   include Shield::BearerLogins::Show
 
-  param page : Int32 = 1
+#   param page : Int32 = 1
 
-  get "/bearer-logins/:bearer_login_id" do
-    html ShowPage, bearer_login: bearer_login, token: token
-  end
-end
+#   get "/bearer-logins/:bearer_login_id" do
+#     html ShowPage, bearer_login: bearer_login
+#   end
+# end
diff --git a/src/actions/bearer_logins/token/show.cr b/src/actions/bearer_logins/token/show.cr
new file mode 100644
index 0000000..55449a9
--- /dev/null
+++ b/src/actions/bearer_logins/token/show.cr
@@ -0,0 +1,9 @@
+class BearerLogins::Token::Show < BrowserAction
+  include Shield::BearerLogins::Token::Show
+
+  param page : Int32 = 1
+
+  get "/bearer-logins/token" do
+    html ShowPage, bearer_login: bearer_login, token: token
+  end
+end
diff --git a/src/actions/current_user/bearer_logins/new.cr b/src/actions/current_user/bearer_logins/new.cr
index e8efafe..c728a91 100644
--- a/src/actions/current_user/bearer_logins/new.cr
+++ b/src/actions/current_user/bearer_logins/new.cr
@@ -2,11 +2,7 @@ class CurrentUser::BearerLogins::New < BrowserAction
   include Shield::CurrentUser::BearerLogins::New
 
   get "/account/bearer-logins/new" do
-    operation = CreateBearerLogin.new(
-      user: user,
-      allowed_scopes: BearerScope.action_scopes.map(&.to_s)
-    )
-
+    operation = CreateBearerLogin.new(user: user)
     html NewPage, operation: operation
   end
 end
diff --git a/src/actions/email_confirmations/show.cr b/src/actions/email_confirmations/show.cr
index dee5e89..419e1b6 100644
--- a/src/actions/email_confirmations/show.cr
+++ b/src/actions/email_confirmations/show.cr
@@ -1,7 +1,7 @@
-class EmailConfirmations::Show < BrowserAction
-  include Shield::EmailConfirmations::Show
+# class EmailConfirmations::Show < BrowserAction
+#   include Shield::EmailConfirmations::Show
 
-  get "/email-confirmations/:token" do
-    run_operation
-  end
-end
+#   get "/email-confirmations/:email_confirmation_id" do
+#     run_operation
+#   end
+# end
diff --git a/src/actions/email_confirmations/token/show.cr b/src/actions/email_confirmations/token/show.cr
new file mode 100644
index 0000000..4024586
--- /dev/null
+++ b/src/actions/email_confirmations/token/show.cr
@@ -0,0 +1,7 @@
+class EmailConfirmations::Token::Show < BrowserAction
+  include Shield::EmailConfirmations::Token::Show
+
+  get "/email-confirmations/token/:token" do
+    run_operation
+  end
+end
diff --git a/src/actions/logins/show.cr b/src/actions/logins/show.cr
new file mode 100644
index 0000000..f5ebea7
--- /dev/null
+++ b/src/actions/logins/show.cr
@@ -0,0 +1,9 @@
+# class Logins::Show < BrowserAction
+#   include Shield::Logins::Show
+
+#   param page : Int32 = 1
+
+#   get "/logins/:login_id" do
+#     html ShowPage, login: login
+#   end
+# end
diff --git a/src/actions/password_resets/show.cr b/src/actions/password_resets/show.cr
index cf3e234..85f3ba8 100644
--- a/src/actions/password_resets/show.cr
+++ b/src/actions/password_resets/show.cr
@@ -1,7 +1,7 @@
-class PasswordResets::Show < BrowserAction
-  include Shield::PasswordResets::Show
+# class PasswordResets::Show < BrowserAction
+#   include Shield::PasswordResets::Show
 
-  get "/password-resets/:token" do
-    run_operation
-  end
-end
+#   get "/password-resets/:password_reset_id" do
+#     html ShowPage, password_reset: password_reset
+#   end
+# end
diff --git a/src/actions/password_resets/token/show.cr b/src/actions/password_resets/token/show.cr
new file mode 100644
index 0000000..fd6178b
--- /dev/null
+++ b/src/actions/password_resets/token/show.cr
@@ -0,0 +1,7 @@
+class PasswordResets::Token::Show < BrowserAction
+  include Shield::PasswordResets::Token::Show
+
+  get "/password-resets/token/:token" do
+    run_operation
+  end
+end
diff --git a/src/pages/bearer_logins/show_page.cr b/src/pages/bearer_logins/show_page.cr
index 7d59a91..4919c23 100644
--- a/src/pages/bearer_logins/show_page.cr
+++ b/src/pages/bearer_logins/show_page.cr
@@ -1,21 +1,13 @@
-struct BearerLogins::ShowPage < MainLayout
-  needs bearer_login : BearerLogin
-  needs token : String
+# struct BearerLogins::ShowPage < MainLayout
+#   needs bearer_login : BearerLogin
 
-  def page_title
-    Rex.t(:"page.bearer_login.show.page_title", name: bearer_login.name)
-  end
+#   def page_title
+#     Rex.t(:"page.bearer_login.show.page_title")
+#   end
 
-  def content
-    h1 Rex.t(:"page.bearer_login.show.main_title", name: bearer_login.name)
+#   def content
+#     h1 Rex.t(:"page.bearer_login.show.main_title")
 
-    para Rex.t(:"page.bearer_login.show.copy_token")
-    pre token
-
-    para do
-      link "[=] #{list_text}", to: Index
-      text " | "
-      link "[+] #{add_new_text}", to: CurrentUser::BearerLogins::New
-    end
-  end
-end
+#     #
+#   end
+# end
diff --git a/src/pages/bearer_logins/token/show_page.cr b/src/pages/bearer_logins/token/show_page.cr
new file mode 100644
index 0000000..3e6639b
--- /dev/null
+++ b/src/pages/bearer_logins/token/show_page.cr
@@ -0,0 +1,21 @@
+struct BearerLogins::Token::ShowPage < MainLayout
+  needs bearer_login : BearerLogin
+  needs token : String
+
+  def page_title
+    Rex.t(:"page.bearer_login.show.page_title", name: bearer_login.name)
+  end
+
+  def content
+    h1 Rex.t(:"page.bearer_login.show.main_title", name: bearer_login.name)
+
+    para Rex.t(:"page.bearer_login.show.copy_token")
+    pre token
+
+    para do
+      link "[=] #{list_text}", to: Index
+      text " | "
+      link "[+] #{add_new_text}", to: CurrentUser::BearerLogins::New
+    end
+  end
+end
diff --git a/src/pages/email_confirmations/show_page.cr b/src/pages/email_confirmations/show_page.cr
new file mode 100644
index 0000000..ee07373
--- /dev/null
+++ b/src/pages/email_confirmations/show_page.cr
@@ -0,0 +1,13 @@
+# struct EmailConfirmations::ShowPage < MainLayout
+#   needs email_confirmation : EmailConfirmation
+
+#   def page_title
+#     Rex.t(:"page.email_confirmation.show.page_title")
+#   end
+
+#   def content
+#     h1 Rex.t(:"page.email_confirmation.show.main_title")
+
+#     #
+#   end
+# end
diff --git a/src/pages/logins/show_page.cr b/src/pages/logins/show_page.cr
new file mode 100644
index 0000000..7a7a749
--- /dev/null
+++ b/src/pages/logins/show_page.cr
@@ -0,0 +1,13 @@
+# struct Logins::ShowPage < MainLayout
+#   needs login : Login
+
+#   def page_title
+#     Rex.t(:"page.login.show.page_title")
+#   end
+
+#   def content
+#     h1 Rex.t(:"page.login.show.main_title")
+
+#     #
+#   end
+# end
diff --git a/src/pages/password_resets/show_page.cr b/src/pages/password_resets/show_page.cr
new file mode 100644
index 0000000..f92d13a
--- /dev/null
+++ b/src/pages/password_resets/show_page.cr
@@ -0,0 +1,13 @@
+# struct PasswordResets::ShowPage < MainLayout
+#   needs password_reset : PasswordReset
+
+#   def page_title
+#     Rex.t(:"page.password_reset.show.page_title")
+#   end
+
+#   def content
+#     h1 Rex.t(:"page.password_reset.show.main_title")
+
+#     #
+#   end
+# end
