diff --git a/.gitignore b/.gitignore
index dbbfe62..a3dfee1 100644
--- a/.gitignore
+++ b/.gitignore
@@ -6,5 +6,6 @@
 *.dwarf
 /.crystal-version
 /.env*
+/.node-version
 /*.lock
 /start_server
diff --git a/config/i18n.cr b/config/i18n.cr
index 1883f38..6b2bcaa 100644
--- a/config/i18n.cr
+++ b/config/i18n.cr
@@ -1,4 +1,4 @@
-I18n.config.loaders << I18n::Loader::YAML.new("config/locales")
+I18n.config.loaders << I18n::Loader::YAML.new("lang")
 I18n.config.default_locale = :en
 I18n.init
 
diff --git a/config/locales/en.yml b/lang/en.yml
similarity index 100%
rename from config/locales/en.yml
rename to lang/en.yml
diff --git a/shard.yml b/shard.yml
index 8e36d6f..0a4a900 100644
--- a/shard.yml
+++ b/shard.yml
@@ -38,7 +38,7 @@ dependencies:
     version: ~> 0.2.1
   shield:
     github: GrottoPress/shield
-    version: ~> 0.20.0
+    version: ~> 0.21.0
   unslash:
     github: GrottoPress/unslash
     version: ~> 0.1.0
